[stable-2.16]  powershell - Improve CLIXML parsing (#83847)

jborean93 opened the pull request:
##### SUMMARY
Improves the logic used when parsing CLIXML to support all escaped character sequences and not just newlines.

Backport of https://github.com/ansible/ansible/pull/83847

(cherry picked from commit b5e0293645570f3f404ad1dbbe5f006956ada0df)

##### ISSUE TYPE
- Bugfix Pull Request

webknjaz commented:
```yaml
fatal: [windows-2022]: FAILED! => {
    "changed": false,
    "elapsed": 0,
    "msg": "Reboot command failed, error was:  The interface is unknown.(1717)\n#< CLIXML\r\n<Objs Version=\"1.1.0.1\" xmlns=\"[http://schemas.microsoft.com/powershell/2004/04\"><Obj](http://schemas.microsoft.com/powershell/2004/04/%22%3E%3CObj) S=\"progress\" RefId=\"0\"><TN RefId=\"0\"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N=\"SourceId\">1</I64><PR N=\"Record\"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>",
    "rebooted": false,
    "start": "2024-08-23T01:23:23.755400+00:00"
}
```
*(https://dev.azure.com/ansible/ansible/_build/results?buildId=121256&view=logs&j=725cd7b8-a3b8-5ca1-c7ff-14bf4c3786cd&t=e6f0fcf2-dc4d-5b13-a1c1-f3795958156f&l=378)*

Was this one supposed to be fixed with this patch?

jborean93 commented:
No that seems to be an unrelated issue, it fails due to the rc not being 0 so the CLIXML filtering is unrelated there.

jborean93 committed:
ssh and psrp - Support more complex characters in fetch_file (#83753) (#83816)

* ssh and psrp - Support more complex chars in fetch_file

Fixes the psrp and ssh (with piped) fetch function to work with paths
that contains glob like characters in the path. For Windows this was
needed when using paths that contain `[]` in the path. For ssh this was
a problem with FreeBSD when using the piped transfer method with similar
characters.

Also tidies up the psrp logic to not inject the paths and buffer size
in the script but pass it as an object through an argument/parameter.

* Fix sanity check

(cherry picked from commit 520fa688ba232ed165fc2dbd3b2be2ebde365ba1)
